mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td

```diff
@@ -167,13 +167,17 @@ def LLVM_SMinOp : LLVM_BinarySameArgsIntrOpI<"smin">;
 def LLVM_UMaxOp : LLVM_BinarySameArgsIntrOpI<"umax">;
 def LLVM_UMinOp : LLVM_BinarySameArgsIntrOpI<"umin">;
 
-class LLVM_MemcpyIntrOpBase<string name> :
+class LLVM_MemcpyIntrOpBase<string name, list<Trait> traits = [],
+                            bit requiresTBAAStruct = 0> :
     LLVM_ZeroResultIntrOp<name, [0, 1, 2],
-    [DeclareOpInterfaceMethods<PromotableMemOpInterface>,
-     DeclareOpInterfaceMethods<DestructurableAccessorOpInterface>,
-     DeclareOpInterfaceMethods<SafeMemorySlotAccessOpInterface>],
+    !listconcat(traits, [
+      DeclareOpInterfaceMethods<PromotableMemOpInterface>,
+      DeclareOpInterfaceMethods<DestructurableAccessorOpInterface>,
+      DeclareOpInterfaceMethods<SafeMemorySlotAccessOpInterface>
+    ]),
     /*requiresAccessGroup=*/1, /*requiresAliasAnalysis=*/1,
-    /*immArgPositions=*/[3], /*immArgAttrNames=*/["isVolatile"]> {
+    /*immArgPositions=*/[3], /*immArgAttrNames=*/["isVolatile"],
+    requiresTBAAStruct> {
```

Comment:
Because of the templated way that `LLVM_IntrOpBase` uses `llvmBuilder`, we are required to add any additional conversion code that gets generated directly to the builder [here](https://github.com/PennyLaneAI/llvm-project/pull/1/files#diff-f5cf42cf7273b3ff2cf4e30626db03a354f787bc28f22957784dd91de3ecb4d8R326).

So we use the `requiresTBAAStruct` to let conditional "injection" of the code into the builder. I believe it would be the same process regardless of how many memory operations use it.

We could remove it by moving the `TBAAStructInterface` methods into the `AliasAnalysisInterface` and then placing this code with the `requiresAliasAnalysis` code generation.

Reply:
Thanks I understand, current implementation is sufficient :+1: